adagrams/game.py

```diff
@@ -1,11 +1,273 @@
+import random
+
 def draw_letters():
-    pass
+    
+    letter_pool_str= "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    # List comp for converting ABC string on line 7 into list 
+    letter_pool_list = [letter for letter in letter_pool_str]
+    
+    # Original letter pool dict - stays constant
+    letter_pool_dict = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    # Tracking container: where the letters(key) and their freq (value) will be stored as separate dictionaries
+    tracking_letter_dict = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+
+    # List of randomly drawn hand, stores 10 letters or tiles
+    hand = []
+    
+    # Adding a random letter to the hand, and reducing frequency for each draw
+    while len(hand) < 10:
+        random_letter = letter_pool_list[random.randint(0, 25)]
+        initial_letter_count = letter_pool_dict[random_letter]
+        random_letter_count = tracking_letter_dict[random_letter]
+        if 0 < random_letter_count <= initial_letter_count:
```

Comment:
This line of could be condense into: 

```python
if `random_letter_count`:
```

This is doing the same thing as your code but just checking to see if there are any leftover occurrences for a letter before we try to append a letter.